src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -268,10 +276,18 @@ private void ConfigureMailboxes()
         /// </summary>
         private void ConfigureProvider()
         {
-            Type providerType = Type.GetType(_settings.ProviderClass);
-            global::System.Diagnostics.Debug.Assert(providerType != null, "providerType != null");
-            var provider = (IActorRefProvider)Activator.CreateInstance(providerType, _name, _settings, _eventStream);
-            _provider = provider;
+            try
+            {
+                Type providerType = Type.GetType(_settings.ProviderClass, true);
+                global::System.Diagnostics.Debug.Assert(providerType != null, "providerType != null");
+                var provider = (IActorRefProvider)Activator.CreateInstance(providerType, _name, _settings, _eventStream);
+                _provider = provider;
+            }
+            catch (TypeLoadException ex)
+            {
+                var msgException = string.Format("Could not find type '{0}'", _settings.ProviderClass);
```

Comment:
We should indicate that the type should be used as an `ActorRefProvider`..
e.g. "Could not find the actorref provider type '{0}'"
We also need to check that the type is assignable to `Akka.Actor.IActorRefProvider`, if not, throw a configuration exception indicating that the type is not a valid actor ref provider.